src/reconcile/rec_write.c

```diff
@@ -1871,21 +1857,11 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
           session, btree->maxleafpage, compressed_size, last_block, &btree->maxleafpage_precomp);
 
 copy_image:
-#ifdef HAVE_DIAGNOSTIC
-    /*
-     * The I/O routines verify all disk images we write, but there are paths in reconciliation that
-     * don't do I/O. Verify those images, too.
-     */
-    WT_ASSERT(session, verify_image == false ||
-        __wt_verify_dsk_image(
-          session, "[reconcile-image]", chunk->image.data, 0, &multi->addr, true) == 0);
-#endif
-
     /*
      * If re-instantiating this page in memory (either because eviction wants to, or because we
      * skipped updates to build the disk image), save a copy of the disk image.
      */
-    if (F_ISSET(r, WT_REC_SCRUB) || r->leave_dirty)
+    if (F_ISSET(r, WT_REC_SCRUB) || (r->cache_write_restore && multi->supd != NULL))
```

Comment:
Same as above: if `cache_write_restore == true`, doesn't `multi->supd` have to be non-NULL?